airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -518,17 +518,54 @@ class SellerFeedbackReports(IncrementalReportsAmazonSPStream):
     Field definitions: https://sellercentral.amazon.com/help/hub/reference/G202125660
     """
 
+    # The list of MarketplaceIds can be found here https://docs.developer.amazonservices.com/en_UK/dev_guide/DG_Endpoints.html
+    MARKETPLACE_DATE_FORMAT_MAP = dict(
+        # eu
+        A2VIGQ35RCS4UG="D/M/YY",  # AE
+        A1PA6795UKMFR9="D/M/YY",  # DE
+        A1C3SOZRARQ6R3="D/M/YY",  # PL
+        ARBP9OOSHTCHU="D/M/YY",  # EG
+        A1RKKUPIHCS9HS="D/M/YY",  # ES
+        A13V1IB3VIYZZH="D/M/YY",  # FR
+        A21TJRUUN4KGV="D/M/YY",  # IN
+        APJ6JRA9NG5V4="D/M/YY",  # IT
+        A1805IZSGTT6HS="D/M/YY",  # NL
+        A17E79C6D8DWNP="D/M/YY",  # SA
+        A2NODRKZP88ZB9="D/M/YY",  # SE
+        A33AVAJ2PDY3EV="D/M/YY",  # TR
+        A1F83G8C2ARO7P="D/M/YY",  # UK
+        # fe
+        A39IBJ37TRP1C6="D/M/YY",  # AU
+        A1VC38T7YXB528="YY/M/D",  # JP
+        A19VAU5U5O7RUS="D/M/YY",  # SG
+        # na
+        ATVPDKIKX0DER="M/D/YY",  # US
+        A2Q3Y263D00KWC="D/M/YY",  # BR
+        A2EUQ1WTGCTBG2="M/D/YY",  # CA
+        A1AM78C64UM0Y8="D/M/YY",  # MX
+    )
+
     name = "GET_SELLER_FEEDBACK_DATA"
     cursor_field = "Date"
     transformer: TypeTransformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization | TransformConfig.CustomSchemaNormalization)
 
-    @transformer.registerCustomTransform
-    def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
-        if original_value and "format" in field_schema and field_schema["format"] == "date":
-            transformed_value = pendulum.from_format(original_value, "M/D/YY").to_date_string()
-            return transformed_value
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.transformer.registerCustomTransform(self.get_transform_function())
+
+    def get_transform_function(self):
+        def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any:
+            if original_value and "format" in field_schema and field_schema["format"] == "date":
+                marketplace_id = self.marketplace_ids[0]
```

Comment:
I'm afraid that your fix won't work if the connector user is using multiple marketplace ids in their connection spec. Do we have a way to link feedback data to a marketplace? At first sight, it does not look possible. I'd suggest the following if you have no other option: return original value if `len(self.marketplace_ids[0]) > 1` and log a warning. If you agree with implementing this solution please update your tests accordingly.

Reply:
@alafanechere I am not sure I understand.
It is now impossible to select more than 1 marketplace for a stream.
It is not possible to select more than 1 in the UI
Also please take a look here:
https://github.com/Labanoras/airbyte/blob/fix-amazon-seller-partner-GET_SELLER_FEEDBACK_DATA/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py#L87-L111

This line, it just creates an array from 1 element.
```python
"marketplace_ids": [marketplace_id],
```

Reply:
Yes, you're right, sorry for my confusion. Don't you think it could make more sense to avoid this kind of confusion to rename the stream attribute to `marketplace_id` typed as a string and only transform it to list in the API call?

Reply:
Yes that makes sense

Reply:
@alafanechere I have refactored confusing plural marketplace ids